db/config.cc

```diff
@@ -427,6 +427,8 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , commitlog_total_space_in_mb(this, "commitlog_total_space_in_mb", value_status::Used, -1,
         "Total space used for commitlogs. If the used space goes above this value, Scylla rounds up to the next nearest segment multiple and flushes memtables to disk for the oldest commitlog segments, removing those log segments. This reduces the amount of data to replay on startup, and prevents infrequently-updated tables from indefinitely keeping commitlog segments. A small total commitlog space tends to cause more flush activity on less-active tables.\n"
         "Related information: Configuring memtable throughput")
+    , commitlog_compression(this, "commitlog_compression", value_status::Used, {}, 
+        "Compression to apply to the commit log. If omitted, the commit log will be written uncompressed.  LZ4, Snappy, and Deflate compressors are supported")
```

Comment:
If ZSTD could be supported - great. Otherwise, I'm not sure Deflate is fast enough for this use case.

Reply:
Actually, it is. I just did not add it to the unit test (nor change that option desc lifted from cassandra). On platforms we've compiled in zstd, setting class_name = ZSTDCompressor will work. 
Zstd is also one of the compressors for which actual fragmented compression/decompression has a much cleaner API. LZ4 is somewhat painful.

Reply:
IIRC zstd compresses better but is much slower than lz4.

Reply:
For the record, I tried adding testing with zstd, but I get weird allocation errors from the zstd calls. Seems to be a config conflict (block size?) between sstable usage and this. Might improve by implementing proper fragmented compress.

Reply:
zstd gave ~50% compression for those cases it worked btw.

Reply:
zstd is great - it's deflate that I feel is not competitive in any way - it doesn't seem to have an advantage of zstd.

Reply:
I'd remove deflate altogether. It's another item we'll need to test, support, etc. - for not good reason.